scrapy_zyte_api/addon.py

```diff
@@ -74,7 +76,7 @@ def update_settings(self, settings: BaseSettings) -> None:
             settings.set(
                 "REQUEST_FINGERPRINTER_CLASS",
                 "scrapy_zyte_api.ScrapyZyteAPIRequestFingerprinter",
-                settings.getpriority("REQUEST_FINGERPRINTER_CLASS"),
+                cast(int, settings.getpriority("REQUEST_FINGERPRINTER_CLASS")),
```

Comment:
is it a Scrapy typing issue that getpriority doesn't annotated as returning int?

Reply:
It can be None. IIRC I decided that a cast is more simple than extracting the var and checking it.

Reply:
Is it an error if the priority is None? Or is it guaranteed we'll get an int? Sorry, I'm a bit confused about the priority :) If it's None, wouldn't settings.set allow it? If no, how would you get None in a first place?

Reply:
`getpriority()` returns None for unset settings.